lib/nexus_parser.rb

```diff
@@ -193,18 +193,45 @@ def update_chr(options = {} )
     @opt.delete(:name)
 
     # the rest have states
-    @opt.keys.each do |k|
+    create_or_update_states_for_character(@index, @opt)
```

Comment:
All these `@` variable reminds me that we need to move to use a better getter/setter pattern, i.e. with `attr_accessor` in various cases.  We shouldn't be getting with `@foo`, we should be using `foo` method as a getter. We should only use `@foo =`.